### ISSUE TITLE
Mono's Vector128 support incorrectly handles Abs for signed integer MinValue

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#95592]
Repro:
https://github.com/dotnet/runtime/pull/95588
```C#
        [Fact]
        public void Vector128SByteAbs_MinValue()
        {
            Vector128<sbyte> vector = Vector128.Create(sbyte.MinValue);
            Vector128<sbyte> abs = Vector128.Abs(vector);
            for (int index = 0; index < Vector128<sbyte>.Count; index++)
            {
                Assert.Equal(sbyte.MinValue, vector.GetElement(index));
            }
        }
```

This test passes on coreclr. On mono it fails with:
```
    System.Runtime.Intrinsics.Tests.Vectors.Vector128Tests.Vector128SByteAbs_MinValue [FAIL]
      System.OverflowException : Negating the minimum value of a twos complement number is invalid.
      Stack Trace:
        /_/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector128Tests.cs(163,0): at System.Runtime.Intrinsics.Tests.Vectors.Vector128Tests.Vector128SByteAbs_MinValue()
        /_/src/mono/System.Private.CoreLib/src/System/Reflection/MethodBaseInvoker.Mono.cs(22,0): at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
        /_/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBaseInvoker.cs(57,0): at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

This is the source of the test failures in https://github.com/dotnet/runtime/pull/94555#issuecomment-1811472249.

cc: @radekdoulik, @tannergooding

**Comment by MichalPetryka** [comment]
https://github.com/dotnet/runtime/blob/4ceefab87d87148fc52499486ce7d6d474384fce/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Scalar.cs#L165

The issue is here, the JIT doesn't expand Abs for integers and uses the managed fallback which is broken so this isn't Mono specific.

**Comment by tannergooding** [comment]
This might be a bug in `Scalar<T>.Abs` which is used for the software fallback. Looks like its just deferring to `Math.Abs` which would throw for `MinValue` rather than returning `MinValue`

**Comment by akoeplinger** [comment]
@lambdageek I saw you added this to 8.x milestone, do we want to backport the fix or is 9.0 ok?

**Comment by lambdageek** [comment]
Probably net9.0 is ok unless we hear from a customer
